pkgs/development/compilers/zig/setup-hook.sh

```diff
@@ -0,0 +1,74 @@
+# shellcheck shell=bash
+
+function zigSetGlobalCacheDir {
+
+    export ZIG_GLOBAL_CACHE_DIR=$(mktemp -d)
+}
+
+function zigBuildPhase {
+    runHook preBuild
+
+    # The local zigDefaultBuildFlags flagset below is meant to avoid CPU feature
+    # impurity in Nixpkgs. However, this flagset is "unstable", because it is
+    # not always exposed by the developer. Because of this, @andrewrk proposed
+    # some solutions that will be implemented in future releases of Zig.
+
+    # Below are some useful links describing the discovery process of this
+    # 'bug' in Nixpkgs:
+
+    # https://github.com/NixOS/nixpkgs/issues/169461
+    # https://github.com/NixOS/nixpkgs/issues/185644
+    # https://github.com/NixOS/nixpkgs/pull/197046
+    # https://github.com/NixOS/nixpkgs/pull/241741#issuecomment-1624227485
+    # https://github.com/ziglang/zig/issues/14281#issuecomment-1624220653
+
+    # Many thanks to the Zig Foundation for their efforts!
+
+    local zigDefaultBuildFlags="-Drelease-safe -Dcpu=baseline"
+
+    zigBuildFlags="$zigDefaultBuildFlags $zigBuildFlags"
+
+    echo "zig build flags: $zigBuildFlags ${zigBuildFlagsArray[@]}"
+
+    zig build $zigBuildFlags "${zigBuildFlagsArray[@]}"
```

Comment:
we should be able to do something like this
```suggestion
    (set -x; zig build $zigBuildFlags "${zigBuildFlagsArray[@]}")
```

Reply:
Good idea!

Reply:
I see that you switched to `echoCmd`, did this approach not work?

Reply:
Well, it worked, however I preferred the traditional approach.